pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -1475,6 +1476,23 @@ public static class MultiStageQueryRunner {
     public static final String KEY_OF_MAX_ROWS_IN_JOIN = "pinot.query.join.max.rows";
     public static final String KEY_OF_JOIN_OVERFLOW_MODE = "pinot.query.join.overflow.mode";
 
+    /// Specifies the send stats mode used in MSE.
+    ///
+    /// Valid values are (in lower or upper case):
+    /// - "SAFE": MSE will only send stats if all instances in the cluster are running 1.4.0 or later.
+    /// - "ALWAYS": MSE will always send stats, regardless of the version of the instances in the cluster.
+    /// - "NEVER": MSE will never send stats.
+    ///
+    /// The reason for this flag that versions 1.3.0 and lower have two undesired behaviors:
+    /// 1. Some queries using intersection generate incorrect stats
+    /// 2. When stats from other nodes are sent but are different from expected, the query fails.
+    ///
+    /// In 1.4.0 the first issue is solved and instead of failing when unexpected stats are received, the query
+    /// continues without children stats. But if a query involves servers in versions 1.3.0 and 1.4.0, the one
+    /// running 1.3.0 may fail, which breaks backward compatibility.
+    public static final String KEY_OF_SEND_STATS_MODE = "pinot.query.mse.stats.mode";
+    public static final String DEFAULT_SEND_STATS_MODE = "safe";
```

Comment:
(minor) To be consistent with the enum
```suggestion
    public static final String DEFAULT_SEND_STATS_MODE = "SAFE";
```